pkg/util/hint/hint.go

```diff
@@ -742,9 +750,7 @@ func (*PlanHints) MatchTableName(tables []*HintedTable, hintTables []HintedTable
 			if table == nil {
 				continue
 			}
-			if (curEntry.DBName.L == table.DBName.L || curEntry.DBName.L == "*") &&
-				curEntry.TblName.L == table.TblName.L &&
-				table.SelectOffset == curEntry.SelectOffset {
+			if curEntry.Match(table) {
```

Comment:
Looks like this function was called by many other functions, like `IfPreferMergeJoin/IfPreferBroadcastJoin...`.

My question is this bug really **only** affect the `read_from_storage` hint?

Reply:
Please see the pic below, other hints are using MatchTableNames, which has already considered crossdb bindings. Only `read_from_storage` has this problem.
<img width="745" height="348" alt="image" src="https://github.com/user-attachments/assets/09719636-bdf1-412c-998e-1512875cd0a9" />